File: src/main/java/no/nav/dokgen/configuration/LoggerInterceptor.java
package no.nav.dokgen.configuration;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;
import org.springframework.web.servlet.ModelAndView;
import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@Component
public class LoggerInterceptor extends HandlerInterceptorAdapter {

    @Override
    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
        LOG.info("[pre-handle]- " + request.getMethod() + " " + request.getRequestURI());
        return super.preHandle(request, response, handler);
    }

    @Override
    public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler,
                           ModelAndView modelAndView) throws Exception {
        LOG.info("[post-handle] - " + request.getMethod() + " " + request.getRequestURI() + " " + response.getStatus());
        super.postHandle(request, response, handler, modelAndView);
    }

    private Logger LOG = LoggerFactory.getLogger(LoggerInterceptor.class);
}

File: src/main/kotlin/no/nav/dokgen/configuration/LoggerInterceptor.kt
package no.nav.dokgen.configuration

import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component
import org.springframework.web.servlet.HandlerInterceptor
import org.springframework.web.servlet.ModelAndView

@Component
class LoggerInterceptor : HandlerInterceptor {
    private val log = LoggerFactory.getLogger(LoggerInterceptor::class.java)

    @Throws(Exception::class)
    override fun preHandle(request: HttpServletRequest, response: HttpServletResponse, handler: Any): Boolean {
        log.info("[pre-handle] - {} {}", request.method, request.requestURI)
        return true
    }

    @Throws(Exception::class)
    override fun postHandle(
        request: HttpServletRequest,
        response: HttpServletResponse,
        handler: Any,
        modelAndView: ModelAndView?
    ) {
        log.info("[post-handle] - {} {} {}", request.method, request.requestURI, response.status)
    }
}